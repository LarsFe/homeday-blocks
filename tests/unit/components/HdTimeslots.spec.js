import { mount } from '@vue/test-utils';
import HdTimeslots from '@/components/HdTimeslots.vue';
import generateSlots from '@/stories/HdTimeslots.stories.js'

describe('HdTimeslots', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = mount(HdTimeslots, {
      timeslots: generateSlots(400, 20),
    });
  });

  test('renders component', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  test('once the component is mounted, the first slot is selected', () => {
    const timeslots = wrapper.findAll('timeslots__slot');

    expect(timeslots[0].classes()).toContain('timeslots__slot--isSelected');
  });
});
