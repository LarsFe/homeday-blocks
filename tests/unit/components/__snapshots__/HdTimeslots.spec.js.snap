// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HdTimeslots renders component 1`] = `
<section class="timeslots">
  <div class="timeslots__controls"><button autocomplete="off" class="arrowButton arrowButton--left" isdisabled="true"></button> <button autocomplete="off" class="arrowButton arrowButton--right"></button></div>
  <div class="timeslots__inner">
    <div class="timeslots__slots" name="timeslots-transition-right"><span class="timeslots__slot timeslots__slot--isSelected">
          11:20
        </span><span class="timeslots__slot">
          11:40
        </span><span class="timeslots__slot">
          12:00
        </span><span class="timeslots__slot">
          12:20
        </span><span class="timeslots__slot">
          12:40
        </span><span class="timeslots__slot">
          13:00
        </span><span class="timeslots__slot">
          13:20
        </span><span class="timeslots__slot">
          13:40
        </span><span class="timeslots__slot">
          14:00
        </span><span class="timeslots__slot">
          14:20
        </span></div>
    <div class="timeslots__legend">
      <p class="timeslots__legend__indicator timeslots__legend__indicator--isUnavailable">
        NOT_AVAILABLE
      </p>
      <p class="timeslots__legend__indicator  timeslots__legend__indicator--isAvailable">
        AVAILABLE
      </p>
    </div>
  </div>
</section>
`;
